src/sdk/namespace_fs.js

```diff
@@ -1527,7 +1534,8 @@ class NamespaceFS {
             const pre_known_size = (params.size >= 0);
             if (!pre_known_size) {
                 // 2.1
-                upload_res = await buffers_pool_sem.surround_count(config.NSFS_BUF_SIZE, async () => this._upload_stream(md_upload_params));
+                const bp = multi_buffer_pool.get_buffers_pool(config.NSFS_BUF_SIZE_L);
+                upload_res = await bp.sem.surround_count(config.NSFS_BUF_SIZE_L, async () => this._upload_stream(md_upload_params));
```

Comment:
we want to avoid reading the config dynamically and instead use the largest pool.
get_buffers_pool(undefined) can be used for that.